examples/standalone/runfile/dynamics.py

```diff
@@ -1,120 +1,152 @@
+import json
+import pathlib
 from argparse import ArgumentParser
+from datetime import datetime
 
-import mpi4py
+import git
+import numpy as np
 import serialbox
 import yaml
+from mpi4py import MPI
 
 import fv3core
 import fv3core._config as spec
 import fv3core.stencils.fv_dynamics as fv_dynamics
 import fv3core.testing
 import fv3gfs.util as util
-from fv3core.testing import GlobalTimer, write_to_json
+from fv3gfs.util import Timer
+
+
+def print_and_write_global_timings(
```

Comment:
It would be worthwhile to split this function into more routines so that it contains no low-level code and only high-level function calls. It has several responsibilities right now:
- Generate a timestamp in two different formats
- generate an experiment timing report dictionary
- save the report dictionary to a json file
- print the report (in a format slightly different from what's saved to disk)
- gather statistics across ranks

The feedback below is just for feedback and not to ask you to change anything here.

This function contains IO nested into a function. In terms of rules of thumb, this ends up breaking the single responsibility principle - you end up with many parts of your code with the "how do I write data" responsibility. A better pattern would be to have a function return a timing report, and print and write it outside of that function. One example where this is necessary is if someone wants to use this routine to write a timing report to Google Cloud. We've run into this kind of thing as an issue a lot on the ML team. I think it's not much of an issue here, and you don't need to change it, mostly good to keep in mind when writing code in the fv3core package itself.